### ISSUE TITLE
Could x-flux run on RTX 3090?

### ISSUE THREAD

**Issue by xingyouxin** [XLabs-AI/x-flux#115]
I try to train x-flux with LoRA on RTX 3090 GPU. But it seems that out of memory.

**Comment by LixDemon** [comment]
I met this error on 3090 GPU. 
NotImplementedError: Cannot copy out of meta tensor; no data! Please use torch.nn.Module.to_empty() instead of torch.nn.Module.to() when moving module from meta to a different device.

**Comment by xingyouxin** [comment]
> I met this error on 3090 GPU. NotImplementedError: Cannot copy out of meta tensor; no data! Please use torch.nn.Module.to_empty() instead of torch.nn.Module.to() when moving module from meta to a different device.

Yes, so do I when using the flux-dev-fp8.safetensors (11 GB). I also tried flux1-dev.safetensors (22 GB), resulting in a memory error.

**Comment by LixDemon** [comment]
> > I met this error on 3090 GPU. NotImplementedError: Cannot copy out of meta tensor; no data! Please use torch.nn.Module.to_empty() instead of torch.nn.Module.to() when moving module from meta to a different device.
> 
> Yes, so do I when using the flux-dev-fp8.safetensors (11 GB). I also tried flux1-dev.safetensors (22 GB), resulting in a memory error.

Have you tried sd_scripts for FLUX Lora training? https://github.com/kohya-ss/sd-scripts/tree/99744af53afcb750b9a64b7efafe51f3f0da8826 I see it works with 24GB VRAM GPUs.

**Comment by xingyouxin** [comment]
> > > I met this error on 3090 GPU. NotImplementedError: Cannot copy out of meta tensor; no data! Please use torch.nn.Module.to_empty() instead of torch.nn.Module.to() when moving module from meta to a different device.
> > 
> > 
> > Yes, so do I when using the flux-dev-fp8.safetensors (11 GB). I also tried flux1-dev.safetensors (22 GB), resulting in a memory error.
> 
> Have you tried sd_scripts for FLUX Lora training? https://github.com/kohya-ss/sd-scripts/tree/99744af53afcb750b9a64b7efafe51f3f0da8826 I see it works with 24GB VRAM GPUs.

Sorry, I didn't. But I have tried ai-toolkit for Flux LoRA training, and tested on web-ui successfully. https://github.com/ostris/ai-toolkit/ Now, I am testing 青龙圣者（Qinglongshengzhe）'s framework. If you are interested in it, don't hesitate to get in touch with me at QQ (593851428). https://www.bilibili.com/video/BV1RW421977P/?spm_id_from=333.788&vd_source=85be3ec5e95fd23384a835c58edf1296 We can do a great job together.

**Comment by raulc0399** [comment]
i did run it on 3090, try with fp8

here with lora and controlnet.


```
python3 x-flux/main.py \
 --prompt "$PROMPT" \
 --image ../imgs/ctrl1024.jpeg \
 --control_type canny \
 --repo_id XLabs-AI/flux-controlnet-canny-v3 \
 --name flux-canny-controlnet-v3.safetensors \
 --use_controlnet \
 **--offload --model_type flux-dev-fp8 \**
 --width 1024 --height 1024  --timestep_to_start_cfg 1 \
 --num_steps 30 --true_gs 4 --guidance 4 \
 --use_lora --lora_weight 1.1 \
 --lora_repo_id XLabs-AI/flux-lora-collection \
 --lora_name realism_lora.safetensors \
 --save_path "$output_path"
```
